Tribler/Policies/BoostingManager.py

```diff
@@ -0,0 +1,676 @@
+                self.torrents[tdict['infohash']]['num_seeders'] = new_seed
+                self.torrents[tdict['infohash']]['num_leechers'] = new_leecher
+                self._logger.info("infohash %s changed s:%d l:%d", infohash_str, new_seed, new_leecher)
+
+    def scrape_trackers(self):
+        """
+        Manually scrape tracker by requesting to tracker manager
+        """
+
+        for infohash, _ in self.torrents.iteritems():
+            # tf_meta = torrent['metainfo']
+
+            # torrent handle
+            # lt_torrent = self.session.lm.ltmgr.get_session().find_torrent(lt.big_number(infohash))
+
+            # check health(seeder/leecher)
+            self.session.lm.torrent_checker.add_gui_request(infohash, True)
+
+        self.session.lm.threadpool.add_task(self.scrape_trackers, self.settings.tracker_interval, "CreditMining_scrape")
+
+    def set_archive(self, source, enable):
+        """
+        setting archive of a particular source. Affect all the torrents in this source
+        """
+        if source in self.boosting_sources:
+            self.boosting_sources[source].archive = enable
+            self._logger.info("Set archive mode for %s to %s", source, enable)
+        else:
+            self._logger.error("Could not set archive mode for unknown source %s", source)
+
+    def start_download(self, torrent):
+        """
+        Start downloading a particular torrent and add it to download list in Tribler
+        """
+        def do_start():
+            """
+            add the actual torrent to the manager to download it later
+            :return:
+            """
+            dscfg = DownloadStartupConfig()
+            dscfg.set_dest_dir(self.settings.credit_mining_path)
+            dscfg.set_safe_seeding(False)
+
+            # just a debug variable
+            tobj = torrent
+
+            preload = tobj.get('preload', False)
+
+            # not using Session.start_download because we need to specify pstate
+            try:
+                assert self.session.get_libtorrent()
+                assert not self.session.lm.download_exists(tobj["metainfo"].get_infohash())
+            except AssertionError, err:
+                self._logger.error("Can't fullfil assertion : %s. Not download %s", err.message,
+                                   hexlify(tobj["metainfo"].get_infohash()))
+                return
+
+            self._logger.info("Starting %s preload %s has pstate %s",
+                              hexlify(tobj["metainfo"].get_infohash()), preload,
+                              True if tobj.get('pstate', None) else False)
+
+            tobj['download'] = self.session.lm.add(tobj['metainfo'], dscfg, pstate=tobj.get('pstate', None),
+                                                   hidden=True, share_mode=not preload, checkpoint_disabled=True)
+            tobj['download'].set_priority(tobj.get('prio', 1))
+
+        self.session.lm.threadpool.add_task_in_thread(do_start, 0)
```

Comment:
Why is the execution of the `do_start` method done on the thread pool? (same question for `do_stop`)